test/lifecycle_module_test.dart

```diff
@@ -57,45 +66,30 @@ class TestLifecycleModule extends LifecycleModule {
     manageStreamController(managedStreamController);
     manageStreamSubscription(managedStreamSubscription);
 
+    var getEventListAdder =
+        (String label) => (LifecycleModule _) => eventList.add(label);
+    var onErrorHandler = (Error error) {};
+
     // Parent module events:
-    willLoad.listen((_) {
-      eventList.add('willLoad');
-    });
-    didLoad.listen((_) {
-      eventList.add('didLoad');
-    });
-    willUnload.listen((_) {
-      eventList.add('willUnload');
-    });
-    didUnload.listen((_) {
-      eventList.add('didUnload');
-    });
-    willSuspend.listen((_) {
-      eventList.add('willSuspend');
-    });
-    didSuspend.listen((_) {
-      eventList.add('didSuspend');
-    });
-    willResume.listen((_) {
-      eventList.add('willResume');
-    });
-    didResume.listen((_) {
-      eventList.add('didResume');
-    });
+    willLoad.listen(getEventListAdder('willLoad'), onError: onErrorHandler);
```

Comment:
I wonder if we should be cleaning these subscriptions up or if that even matters?